test/e2e/tests/grpc_test.go

```diff
@@ -30,3 +30,88 @@ func TestGRPC_Version(t *testing.T) {
 		require.Equal(t, version.BlockProtocol, res.Block)
 	})
 }
+
+func TestGRPC_Block_GetByHeight(t *testing.T) {
+	testNode(t, func(t *testing.T, node e2e.Node) {
+		if node.Mode != e2e.ModeFull && node.Mode != e2e.ModeValidator {
+			return
+		}
+
+		blocks := fetchBlockChain(t)
+
+		client, err := node.Client()
+		require.NoError(t, err)
+		status, err := client.Status(ctx)
+		require.NoError(t, err)
+
+		first := status.SyncInfo.EarliestBlockHeight
+		last := status.SyncInfo.LatestBlockHeight
+		if node.RetainBlocks > 0 {
+			first++ // avoid race conditions with block pruning
+		}
+
+		ctx, ctxCancel := context.WithTimeout(context.Background(), time.Minute)
+		defer ctxCancel()
+		gRPCClient, err := node.GRPCClient(ctx)
+		require.NoError(t, err)
+
+		for _, block := range blocks {
+			if block.Header.Height < first {
+				continue
+			}
+			if block.Header.Height > last {
+				break
+			}
+
+			// Get first block
+			firstBlock, err := gRPCClient.GetBlockByHeight(ctx, first)
+
+			// First block tests
+			require.NoError(t, err)
+			require.NotNil(t, firstBlock.BlockID)
+			require.Equal(t, firstBlock.Block.Height, first)
+
+			// Get last block
+			lastBlock, err := gRPCClient.GetBlockByHeight(ctx, last)
+
+			// Last block tests
+			require.NoError(t, err)
+			require.NotNil(t, lastBlock.BlockID)
+			require.Equal(t, lastBlock.Block.Height, last)
+		}
+	})
+}
+
+func TestGRPC_Block_GetLatestHeight(t *testing.T) {
+
+	testnet := loadTestnet(t)
+	node := testnet.RandomNode()
+	if node.Mode != e2e.ModeFull && node.Mode != e2e.ModeValidator {
+		return
+	}
```

Comment:
This just skips the test entirely if the random node happens to be a light client - not sure that's super useful.